source/ShipModel.cpp

```diff
@@ -82,7 +82,29 @@ bool ShipModel::createAllTask(int data) {
 
 bool ShipModel::failAllTask() {
 	for (int i = 0; i < donuts.size(); i++) {
-		float angle = (float)(rand() % (int)(getSize()));
+		float angle;
+		bool goodAngle = false;
+		while (!goodAngle) {
+			angle = (float)(rand() % (int)(getSize()));
+			for (unsigned int k = 0; k < breaches.size(); k++) {
+				float breachAngle = breaches[k]->getAngle();
+				float diff = shipSize / 2 - abs(abs(breachAngle - angle) - shipSize / 2);
+				if (breachAngle == -1 || diff > 15) {
+					goodAngle = true;
+				} else {
+					goodAngle = false;
+				}
+			}
+			for (unsigned int k = 0; k < doors.size(); k++) {
+				float doorAngle = doors[k]->getAngle();
+				float diff = shipSize / 2 - abs(abs(doorAngle - angle) - shipSize / 2);
+				if (doorAngle == -1 || diff > 15) {
+					goodAngle = true;
+				} else {
```

Comment:
I'm confused by this logic. If the breach pass sets goodAngle to false, but the door pass sets goodAngle to true, shouldn't it still be false by the end of it? Likewise, if a bunch of breaches set goodAngle to false but the last breach sets it to true, shouldn't it still be false by the end?

Reply:
@asaxena5 could we get an update abt this PR?

Reply:
@wenjiazhang @mt-xing updated it, I forgot to account for that case